vowpalwabbit/global_data.h

```diff
@@ -311,6 +312,28 @@ struct vw_logger
   vw_logger& operator=(const vw_logger& other) = delete;
 };
 
+struct buffer_restore

```

Comment:
`buffer_replace` instead of `buffer_restore`?

Reply:
done